src/lib/array.jl

```diff
@@ -387,6 +387,19 @@ x::TrackedVector  * y::AbstractVector = track(*, x, y)
 x::AbstractVector * y::TrackedVector  = track(*, x, y)
 x::TrackedVector  * y::TrackedVector  = track(*, x, y)
 
+# Ambiguity fixes
+Base.:*(x::Transpose{T,<:AbstractVector{T}},y::TrackedMatrix) where {T} = track(*, x, y)
```

Comment:
Is `Transpose{AbstractVector}` strictly necessary here, as opposed to `Transpose`? Do we need to have the type parameter?

Would be good to have a test to demonstrate that if so.

Reply:
Yes it is necessary, otherwise if the dimension is unknown then it gets an ambiguity against the previous lines because those specify dimension. The test case I have isn't too minimal, so if you could quickly make one that would be great.

Reply:
Presumably, you only need to multiply a transpose by a tracked matrix to hit these kinds of cases; that shouldn't be too hard to write down.

Reply:
I agree. Can you write it? It's hard to write it from my phone and I'm travelling.

Reply:
Yes, it's strictly necessary.